Modules/App/ElementTemplateHelper/ElementTemplateHelper.cs

```diff
@@ -0,0 +1,152 @@
+using Common.Controls;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using Vixen.Rule;
+using Vixen.Sys;
+using VixenModules.OutputFilter.DimmingCurve;
+using VixenModules.Property.Color;
+
+namespace VixenModules.App.ElementTemplateHelper
+{
+	/// <summary>
+	/// Helper class for processing an element template.
+	/// </summary>
+	public class ElementTemplateHelper
+	{
+		#region Public Methods
+
+		/// <summary>
+		/// Processes an element template to create nodes.
+		/// </summary>
+		/// <param name="selectedNodes">Selected nodes</param>
+		/// <param name="template">Template to apply</param>
+		/// <param name="owner">Window owner</param>
+		/// <param name="addToTree">Delegate for adding create node to the tree</param>
+		/// <param name="treeElements">Element tree to add to</param>
+		/// <returns>Collection of created nodes</returns>
+		public async Task<IEnumerable<ElementNode>> ProcessElementTemplate(
+			IEnumerable<ElementNode> selectedNodes, 
+			IElementTemplate template, 
+			IWin32Window owner,
+			Action<ElementNode> addToTree,
+			ElementTree treeElements)
+		{
+			// Default the created nodes to null
+			IEnumerable<ElementNode> createdElements = null;
+
+			// Perform (optional) setup
+			bool success = template.SetupTemplate(selectedNodes);
+
+			// If the setup was not cancelled then...
+			if (success)
+			{
+				// Generate the nodes
+				createdElements = await template.GenerateElements(selectedNodes);
+
+				// If the user has not cancelled the template then...
+				if (!template.Cancelled)
+				{
+					// If elements were not created then...
+					if (createdElements == null || !createdElements.Any())
+					{
+						//messageBox Arguments are (Text, Title, No Button Visible, Cancel Button Visible)						
+						var messageBox = new MessageBoxForm("Could not create elements.  Ensure you use a valid name and try again.", "",
+							MessageBoxButtons.OKCancel, SystemIcons.Error);
+						messageBox.ShowDialog();
```

Comment:
Should owner be used here in the show dialog to create the parent for the message box? Looks it was not there in the original code, but since it is being refactored, might as well fix that omission.

Reply:
Updated the MesageBox call to specify an owner.  Also included a fix to a bug I encountered.